core/src/test/scala/unit/kafka/server/BrokerLifecycleManagerTest.scala

```diff
@@ -197,11 +197,17 @@ class BrokerLifecycleManagerTest {
     result
   }
 
-  def poll[T](context: RegistrationTestContext, manager: BrokerLifecycleManager, future: Future[T]): T = {
-    while (!future.isDone || context.mockClient.hasInFlightRequests) {
-      context.poll()
+  def poll[T](ctx: RegistrationTestContext, manager: BrokerLifecycleManager, future: Future[T]): T = {
+    while (ctx.mockChannelManager.unsentQueue.isEmpty) {
+      // If the manager is idling until scheduled events we need to advance the clock
+      if (manager.eventQueue.scheduledAfterIdling()
+        .filter(!_.getClass.getSimpleName.endsWith("TimeoutEvent")) // avoid triggering timeout events
```

Comment:
This is a bit ugly. It only applies to `RegistrationTimeoutEvent`. We want to avoid triggering those deferred events.

Instead of checking for that class specifically, I thought there may be other timeout event types in the future and we'll also want to avoid those. I also ruled out extending EventQueue.Event as we only care about this in this test suite.

I feel ambivalent about the approach I've taken here, so please let me know if you have a preference.

Reply:
I see. Perhaps we could add a step after `poll(ctx, manager, registration)` to explicitly drain `RegistrationTimeoutEvent` first? This makes the test safer since it won't be sensitive to the timeout config values.

Reply:
That's a good idea. I'm making that change.